src/main/java/com/techconative/restel/core/managers/RestelDefinitionManager.java

```diff
@@ -117,7 +117,11 @@ private boolean executeTestMethod(
   }
 
   private void validateStatus(RESTResponse response, RestelTestMethod restelTestMethod) {
-    if (!restelTestMethod.getAcceptedStatusCodes().contains(response.getStatus())) {
+    restelTestMethod.setAcceptedStatusCodes(
```

Comment:
We don't have to set this. When we set the resolved value to the method, it'll overwrite any existing variables in it. And if the `TestMethod` is being used by some other scenario, it would be a problem.

And general rule, avoid having a method which modifies the entities that it is not creating. For example, the `restelTestMethod` is not created by `validateStatus` and being passed by some other method. So `restelTestMethod` should refrain from making changes to it. 

On side notes, this is part of nature of immutable data-structures which is common in Functional programming. [Java Record](https://docs.oracle.com/en/java/javase/14/language/records.html) is a similar effort from Java side to keep beans immutable and avoid such mutations that we see here.

Reply:
Thanks for the pointers. Will follow this.